pkg/proc/gdbserial/gdbserver_conn.go

```diff
@@ -1235,24 +1208,9 @@ func (conn *gdbConn) memoryRegionInfo(addr uint64) (*memoryRegionInfo, error) {
 
 	mri := &memoryRegionInfo{}
 
-	buf := resp
-	for len(buf) > 0 {
-		colon := bytes.Index(buf, []byte{':'})
-		if colon < 0 {
-			break
-		}
-		key := buf[:colon]
-		buf = buf[colon+1:]
-
-		semicolon := bytes.Index(buf, []byte{';'})
-		var value []byte
-		if semicolon < 0 {
-			value = buf
-			buf = nil
-		} else {
-			value = buf[:semicolon]
-			buf = buf[semicolon+1:]
-		}
+	csp := colonSemicolonParser{buf: resp}
+	for csp.next() {
+		key, value := csp.key, csp.value
```

Comment:
I propose converting to `string` here and remove convertions below:
```suggestion
		key, value := string(csp.key), string(csp.value)
```

This will make the code more clear.

Reply:
The same comment applies to other usages of `key, value`.

Reply:
It's worse. First in some cases value isn't used as a string. Second, if you cast csp.key to string here it will compile to an unconditional call to runtime.slicebytetostring whereas if it's done in the header of the switch it won't.